src/software/world/world.cpp

```diff
@@ -126,19 +126,32 @@ void World::updateRefboxGameState(const RefboxGameState &game_state)
                         return gamestate == refbox_game_state_history.front();
                     }))
     {
-        game_state_.updateRefboxGameState(game_state);
-        game_state_.updateBall(ball_);
+        current_refbox_game_state_.updateRefboxGameState(game_state);
+        current_refbox_game_state_.updateBall(ball_);
     }
     else
     {
-        game_state_.updateRefboxGameState(game_state_.getRefboxGameState());
-        game_state_.updateBall(ball_);
+        current_refbox_game_state_.updateRefboxGameState(
+            current_refbox_game_state_.getRefboxGameState());
+        current_refbox_game_state_.updateBall(ball_);
     }
 }
 
-void World::updateRefboxData(const RefboxData &refbox_data)
+void World::updateRefboxStage(const RefboxStage &stage)
 {
-    updateRefboxGameState(refbox_data.getGameState());
+    refbox_stage_history.push_back(stage);
+    // Take the consensus of the previous refbox messages
+    if (!refbox_stage_history.empty() &&
+        std::all_of(
+            refbox_stage_history.begin(), refbox_stage_history.end(),
+            [&](auto gamestate) { return gamestate == refbox_stage_history.front(); }))
+    {
+        current_refbox_stage_ = stage;
+    }
+    else
+    {
+        current_refbox_stage_ = stage;
+    }
```

Comment:
maybe im not seeing the hidden galaxy brain part correctly, but why do we need this check? 

Looks like in both cases we just do `current_refbox_stage_ = stage`, can we do without the if/else and just have that one line

Reply:
oof galaxy brain hurts